### PR TITLE
c2cpg: more test refactorings / clean-ups

### PR DESCRIPTION
**Pull request by max-leuthaeuser**
- cleaner type hierarchy for test suites
 - removed unused imports
 - re-used `NamespaceTraversal.globalNamespaceName`

### CODE REVIEW

File: joern-cli/frontends/c2cpg/src/test/scala/io/joern/c2cpg/EnumTests.scala [hunk 1]
```diff
@@ -48,26 +49,27 @@ class EnumTests extends CCodeToCpgSuite(fileSuffix = FileDefaults.CPP_EXT) {
...
-          green.name shouldBe "green"
-          blue.name shouldBe "blue"
-        }
-        inside(color.astChildren.isMethod.l) { case List(sinit) =>
-          sinit.name shouldBe "<sinit>"
-          sinit.fullName shouldBe "color:<sinit>"
-          sinit.code shouldBe "green = 20"
-          inside(sinit.ast.isCall.l) { case List(greenInit) =>
-            greenInit.code shouldBe "green = 20"
+      inside(cpg.typeDecl.nameNot(NamespaceTraversal.globalNamespaceName).filter(x => !x.isExternal).l) {
```
Comment (mpollmeier) [inline joern-cli/frontends/c2cpg/src/test/scala/io/joern/c2cpg/EnumTests.scala:52]:
```suggestion
      inside(cpg.typeDecl.nameNot(NamespaceTraversal.globalNamespaceName).internal.l) {
```

File: joern-cli/frontends/c2cpg/src/test/scala/io/joern/c2cpg/TemplateTests.scala [hunk 2]
```diff
@@ -15,19 +16,20 @@ class TemplateTests extends CCodeToCpgSuite(fileSuffix = FileDefaults.CPP_EXT) {
...
-        y.name shouldBe "Y"
-        y.fullName shouldBe "Y"
-        y.aliasTypeFullName shouldBe Some("Y<A,B>")
-        a.name shouldBe "A"
-        a.fullName shouldBe "A"
-        a.aliasTypeFullName shouldBe Some("X<int>")
-        b.name shouldBe "B"
-        b.fullName shouldBe "B"
-        b.aliasTypeFullName shouldBe Some("Y<int, char>")
+      inside(cpg.typeDecl.nameNot(NamespaceTraversal.globalNamespaceName).filter(x => !x.isExternal).l) {
```
Comment (mpollmeier) [inline joern-cli/frontends/c2cpg/src/test/scala/io/joern/c2cpg/TemplateTests.scala:19]:
```suggestion
      inside(cpg.typeDecl.nameNot(NamespaceTraversal.globalNamespaceName).internal.l) {
```

File: joern-cli/frontends/c2cpg/src/test/scala/io/joern/c2cpg/TemplateTests.scala [hunk 3]
```diff
@@ -36,25 +38,27 @@ class TemplateTests extends CCodeToCpgSuite(fileSuffix = FileDefaults.CPP_EXT) {
...
 
     "be correct for struct templates" in {
       val cpg = code("""
         |template<typename A, typename B> struct Foo;
         |""".stripMargin)
-      inside(cpg.typeDecl.nameNot("<global>").filter(x => !x.isExternal).l) { case List(foo) =>
-        foo.name shouldBe "Foo"
-        foo.fullName shouldBe "Foo"
-        foo.aliasTypeFullName shouldBe Some("Foo<A,B>")
+      inside(cpg.typeDecl.nameNot(NamespaceTraversal.globalNamespaceName).filter(x => !x.isExternal).l) {
```
Comment (mpollmeier) [inline joern-cli/frontends/c2cpg/src/test/scala/io/joern/c2cpg/TemplateTests.scala:57]:
```suggestion
      inside(cpg.typeDecl.nameNot(NamespaceTraversal.globalNamespaceName).internal.l) {
```

File: joern-cli/frontends/c2cpg/src/test/scala/io/joern/c2cpg/TemplateTests.scala [hunk 4]
```diff
@@ -36,25 +38,27 @@ class TemplateTests extends CCodeToCpgSuite(fileSuffix = FileDefaults.CPP_EXT) {
...
         |""".stripMargin)
-      inside(cpg.typeDecl.nameNot("<global>").filter(x => !x.isExternal).l) { case List(x, y) =>
-        x.name shouldBe "X"
-        x.fullName shouldBe "X"
-        x.aliasTypeFullName shouldBe Some("X<T>")
-        y.name shouldBe "Y"
-        y.fullName shouldBe "Y"
-        y.aliasTypeFullName shouldBe Some("Y<A,B>")
-        y.inheritsFromTypeFullName shouldBe Seq("X<A>")
+      inside(cpg.typeDecl.nameNot(NamespaceTraversal.globalNamespaceName).filter(x => !x.isExternal).l) {
```
Comment (mpollmeier) [inline joern-cli/frontends/c2cpg/src/test/scala/io/joern/c2cpg/TemplateTests.scala:41]:
```suggestion
      inside(cpg.typeDecl.nameNot(NamespaceTraversal.globalNamespaceName).internal.l) {
```

**Review by mpollmeier — APPROVED** [review]

File: joern-cli/frontends/c2cpg/src/test/scala/io/joern/c2cpg/EnumTests.scala (hunk 1 above)
Comment (max-leuthaeuser) [reply joern-cli/frontends/c2cpg/src/test/scala/io/joern/c2cpg/EnumTests.scala:52]: Sadly, this does not work here as `.internal` is defined as `canonicalType.has(Properties.IS_EXTERNAL -> false)`.
The canonical type is artificially created from the type alias and is external.

Comment (mpollmeier) [reply joern-cli/frontends/c2cpg/src/test/scala/io/joern/c2cpg/EnumTests.scala:52]: oh, ok that's good to know then...

**Pull request merged**